flutter_app/lib/widgets/nav_bar_widget.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+
+class NavBar extends StatefulWidget {
+  const NavBar({super.key});
+
+  @override
+  NavBarState createState() => NavBarState();
+}
+
+class NavBarState extends State<NavBar> {
+  int _index = 0;
+
+  void _onItemClicked(int index) {
+    setState(() {
+      _index = index;
+    });
+
+    switch (index) {
+      case 0:
+        Navigator.of(context).popUntil((route) => route.isFirst);
```

Comment:
Not really a fan of routing like this, use `pushReplacementNamed` and `popAndPushNamed` when possible or replace the router as suggested above.